jupyterhub/app.py

```diff
@@ -365,12 +366,36 @@ def _log_format_default(self):
         """override default log format to include time"""
         return "%(color)s[%(levelname)1.1s %(asctime)s.%(msecs).03d %(name)s %(module)s:%(lineno)d]%(end_color)s %(message)s"
 
+    extra_log_file = Unicode(
+        "",
+        config=True,
+        help="Set a logging.FileHandler on this file."
+    )
+    extra_log_handlers = List(
+        Instance(logging.Handler),
+        config=True,
+        help="Extra log handlers to set on JupyterHub logger",
+    )
+
     def init_logging(self):
         # This prevents double log messages because tornado use a root logger that
         # self.log is a child of. The logging module dipatches log messages to a log
         # and all of its ancenstors until propagate is set to False.
         self.log.propagate = False
-        
+
+        if self.extra_log_file:
+            self.extra_log_handlers.append(
+                logging.FileHandler(self.extra_log_file)
+            )
+
+        _formatter = self._log_formatter_cls(
+            fmt=self.log_format,
+            datefmt=self.log_datefmt,
+        )
+        for handler in self.extra_log_handlers:
+            handler.setFormatter(_formatter)
```

Comment:
should we always be setting the log formatter, or trusting that they are already configured with their own formatter?